src/hex/coordinate.rs

```diff
@@ -729,4 +1023,25 @@ mod tests {
         assert_eq!(axial!(0, 0).rotate(Some(axial!(1, 1)), 2), axial!(3, 0));
         assert_eq!(axial!(0, 0).rotate(Some(axial!(1, 1)), 3), axial!(2, 2));
     }
+
+    #[test]
+    fn shared_vert_two() {
+        assert_eq!(
+            axial!(1, 1).shared_vert_two(axial!(2, 0)).unwrap(),
+            [
+                vertex!(2, 0, VertexSpin::Down),
+                vertex!(1, 1, VertexSpin::Up)
+            ]
+        );
+    }
+
+    #[test]
+    fn shared_vert_three() {
+        assert_eq!(
+            axial!(1, 1)
+                .shared_vert_three(axial!(2, 0), axial!(2, 1))
+                .unwrap(),
+            vertex!(2, 0, VertexSpin::Down)
+        );
+    }
```

Comment:
As this file is over 1000 lines long, we may wish to consider splitting it back up.

Reply:
Separated out vertex logic into vertex file.